src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,49 +1,104 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int ARRAY_SIZE = 10;
+    private Object[] array;
+    private int counter;
+
+    public ArrayList() {
+        array = new Object[ARRAY_SIZE];
+        counter = 0;
+    }
 
     @Override
     public void add(T value) {
-
+        growIfRequire();
+        array[counter] = value;
+        counter = counter + 1;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index == counter) {
+            add(value);
+        } else {
+            indexValidator(index);
+            growIfRequire();
+            System.arraycopy(array, index, array, index + 1, counter - index);
+            array[index] = value;
+            counter = counter + 1;
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        indexValidator(index);
+        return (T) array[index];
     }
 
     @Override
     public void set(T value, int index) {
-
+        indexValidator(index);
+        array[index] = value;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        indexValidator(index);
+        T oldValue = get(index);
+        System.arraycopy(array, index + 1, array, index, counter - index - 1);
+        counter = counter - 1;
+        return oldValue;
     }
 
     @Override
     public T remove(T t) {
-        return null;
+        for (int i = 0; i < counter; i++) {
+            if (array[i] == null || array[i].equals(t)) {
```

Comment:
let's refactor this line - check how we compared two objects in the previous solutions